### PR TITLE
Verify bypass behavior under different traffic profiles v3

### PR DESCRIPTION
**Pull request by lukashino**
Follow-up of https://github.com/OISF/suricata-verify/pull/2078

Redmine ticket:
https://redmine.openinfosecfoundation.org/issues/6788

Note:
- min-version changes between tests between 7 and 8, the ones with min-version == 7 are where behavior is not expected to change, and the others are where the behavior changes (TLS bypass decoupling, SSH bypass). 

Describe changes:
v3:
- rebased

### CODE REVIEW

File: tests/ssh-hassh/test.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ features:
     - RUST
 
 args:
- - -k none --set stream.bypass=yes
+ - -k none --set stream.bypass=yes --set app-layer.protocols.ssh.encryption-handling=bypass
```
Comment (catenacyber) [inline tests/ssh-hassh/test.yaml:7]: Why do we need this change ?

File: tests/bypass-tls-enabled/test.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+requires:
+    min-version: 8
+
```
Comment (catenacyber) [inline tests/bypass-tls-enabled/test.yaml:3]: Is it reusing a pcap ?

File: tests/ssh-hassh/test.yaml (hunk 1 above)
Comment (lukashino) [reply tests/ssh-hassh/test.yaml:7]: behavior change

- Encrypted SSH traffic bypass is now independently controlled through
  `app-layer.protocols.ssh.encryption-handling` setting. The setting can either
  be `bypass`, `default` or `full`. 
  To retain the previous behavior of encrypted traffic bypass
  combined with stream depth bypass, set
  `app-layer.protocols.ssh.encryption-handling` to `bypass` (while also
  setting `app-layer.protocols.tls.encryption-handling` to `bypass` and
  `stream.bypass` to `true`).

Previously SSH was bypassed after the session turned encrypted, now we need to be more explicit.

**Comment by lukashino** [comment]
cont in https://github.com/OISF/suricata-verify/pull/2170

P's comment is mentioned in the PR

File: tests/ssh-hassh/test.yaml (hunk 1 above)
Comment (lukashino) [reply tests/ssh-hassh/test.yaml:7]: Re 
> Previously SSH was bypassed after the session turned encrypted, now we need to be more explicit.

So previously when the SSH session turned encrypted and stream bypass was allowed then it bypassed the flow. This was not related to reaching the stream depth but only to the session state. 
Currently, you can replicate the behavior with setting encryption handling to bypass.

**Pull request closed without merging**